src/subdevice/mod.rs

```diff
@@ -776,6 +786,11 @@ impl<'a, S> SubDeviceRef<'a, S> {
         self.configured_address
     }
 
+    /// Get the index of this subdevice in the overall EtherCAT network when it was first initialized.
+    pub fn topology_address_at_init(&self) -> u16 {
```

Comment:
Rather than rely on ethercrab continuing to configure addresses in the future based on a fixed base plus the position in the topology, I figured I'd add that information to the API explicitly.